adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/JaxRsImplementationCodeGenerator.java

```diff
@@ -76,34 +76,34 @@ private boolean containsResourcesOfType(final JDefinedClass resourceInterface, f
         return resourceInterface.methods().stream().filter((interfaceMethod) -> isResourceOfType(interfaceMethod, methodType)).findAny().isPresent();
     }
 
-
-    private void addAnnotations(final JDefinedClass resourceImplementation) {
+    private void addAnnotations(final JDefinedClass resourceImplementation, final Component component) {
         resourceImplementation.annotate(Stateless.class);
-        JAnnotationUse adapterAnnotation = resourceImplementation.annotate(Adapter.class);
-        adapterAnnotation.param("value", Component.COMMAND_API);
+        final JAnnotationUse adapterAnnotation = resourceImplementation.annotate(Adapter.class);
+
+        adapterAnnotation.param("value", component);
     }
 
-    private void addImplementationMethods(final JDefinedClass resourceImplementation, JDefinedClass resourceInterface) {
-        JClass str = codeModel.ref("String");
+    private void addImplementationMethods(final JDefinedClass resourceImplementation, final JDefinedClass resourceInterface) {
+        final JClass str = codeModel.ref("String");
         resourceInterface.methods().forEach(interfaceMethod -> {
-            JMethod implementationMethod = resourceImplementation.method(PUBLIC, interfaceMethod.type(),
+            final JMethod implementationMethod = resourceImplementation.method(PUBLIC, interfaceMethod.type(),
                     interfaceMethod.name());
             implementationMethod.annotate(Override.class);
             addMethodParams(implementationMethod, interfaceMethod);
-            JBlock body = implementationMethod.body();
-            JType map = codeModel.ref(Map.class.getCanonicalName()).narrow(str, str);
-            JClass mapBuilderClass = codeModel.ref(ImmutableMap.Builder.class.getName()).narrow(str, str);
+            final JBlock body = implementationMethod.body();
+            final JType map = codeModel.ref(Map.class.getCanonicalName()).narrow(str, str);
+            final JClass mapBuilderClass = codeModel.ref(ImmutableMap.Builder.class.getName()).narrow(str, str);
 
             if (isResourceOfType(interfaceMethod, GET)) {
-                body.decl(map, PATH_PARAMS_ARGUMENT_NAME, pathParamsMapBuilderInvocation(mapBuilderClass,
+                body.decl(map, PARAMS_ARGUMENT_NAME, paramsMapBuilderInvocation(mapBuilderClass,
                         implementationMethod, false));
                 body.directStatement(format("return %s.processSynchronously(%s::dispatch, %s, %s);",
-                        REST_PROCESSOR_BEAN_NAME, SYNC_DISPATCHER_BEAN_NAME, HEADERS_CONTEXT, PATH_PARAMS_ARGUMENT_NAME));
+                        REST_PROCESSOR_BEAN_NAME, SYNC_DISPATCHER_BEAN_NAME, HEADERS_CONTEXT, PARAMS_ARGUMENT_NAME));
             } else {
```

Comment:
implies we don't have PUT or DELETE etc - should we specifically do this only for POST?